src/v/storage/tests/compaction_e2e_test.cc

```diff
@@ -313,13 +321,17 @@ TEST_F(CompactionFixtureTest, TestRecompactWithNewData) {
     ASSERT_EQ(segments_compacted + 3, segments_compacted_3);
 
     // Check for a reasonable compaction ratio.
-    ASSERT_LT(compaction_ratio_3, 0.5);
+    ASSERT_LT(compaction_ratio_3, 0.65);
```

Comment:
Note for reviewers: as the records per batch increases, with a fixed number of total records, the overall size per record decreases since there are fewer headers, etc. So in increasing the number of records per batch, this value went up a bit.